python/lsst/ts/externalscripts/maintel/make_comcam_bias.py

```diff
@@ -0,0 +1,163 @@
+# This file is part of ts_externalscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["MakeComCamBias"]
+
+import json
+import yaml
+import os
+
+from ..base_make_bias import BaseMakeBias
+from lsst.ts import salobj
+from lsst.ts.observatory.control.maintel.comcam import ComCam
+
+
+class MakeComCamBias(BaseMakeBias):
+    """ Take biases and construct a master bias SAL Script.
+
+    This class takes biases with LSSTComCam and constructs
+    a master bias calling the bias pipetask via OCPS.
+    """
+
+    def __init__(self, index=1):
+        super().__init__(
+            index=index,
+            descr="This class takes biases with LSSTComCam and constructs "
+                  "a master bias calling the bias pipetask via OCPS.",
+        )
+        self.comcam = ComCam(domain=self.domain, log=self.log)
+        self.ocps = salobj.Remote(domain=self.domain, name="OCPS")
+
+    @classmethod
+    def get_schema(cls):
+        schema = """
+        $schema: http://json-schema.org/draft-07/schema#
+        $id: https://github.com/lsst-ts/ts_externalscripts/blob/master/python/lsst/ts/externalscripts/maintel/make_comcam_bias.py
+        title: MakeComCamBias v1
+        description: Configuration for making a LSSTComCam master bias SAL Script.
+        type: object
+        """
+        return yaml.safe_load(schema)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config: `types.SimpleNamespace`
+            Configuration data. See `get_schema` for information about data
+            structure.
+        """
+        # Log information about the configuration
+
+        self.log.debug(
+            f"n_bias: {config.n_bias}, detectors: {config.detectors}, "
+            f"instrument: LSSTComCam. "
+        )
+
+        self.config = config
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script.
+        """
+        # Temporary number
+        metadata.duration = 10
+
+    async def arun(self, checkpoint=False):
```

Comment:
I think this method can be moved to the parent class with some small modifications. 

For instance, if you create the abstract property `camera`, replace all instances of `self.comcam` by `self.camera`.

Then, you can add additional abstract properties for all special attributes, e.g.;

```
data_query=f"instrument='LSSTComCam' AND"
                        f" detector IN {self.config.detectors} AND exposure IN {exposures}"
```

```
REPO = "/repo/LSSTComCam"
```

you will also need a way to differentiate the archivers:

```
self.comcam.rem.ccarchiver.evt_imageInOODS.aget(timeout=20)
```

could be;

```
self.image_in_oods.aget(timeout=20)
```

where image_in_oods is an abstract property that points to

```
self.comcam.rem.<archiver>.evt_imageInOODS
```